peer/network.go

```diff
@@ -378,6 +389,11 @@ func (n *network) AppResponse(_ context.Context, nodeID ids.NodeID, requestID ui
 
 	handler, exists := n.markRequestFulfilled(requestID)
 	if !exists {
```

Comment:
Need to check if this request id never has issues w/ the sdk. It's really dangerous if we do since we can panic on the semaphore if there's a bug here.